task/oci-copy-oci-ta/0.1/oci-copy-oci-ta.yaml

```diff
@@ -0,0 +1,161 @@
+
+          echo "declare OCI_SOURCE=${source}" > /var/workdir/vars/$filename
+          echo "declare OCI_FILENAME=${filename}" >> /var/workdir/vars/$filename
+          echo "declare OCI_ARTIFACT_TYPE=${artifact_type}" >> /var/workdir/vars/$filename
+          echo "declare OCI_ARTIFACT_DIGEST=${artifact_digest}" >> /var/workdir/vars/$filename
+
+          echo "Wrote /var/workdir/vars/$filename with contents:"
+          cat /var/workdir/vars/$filename
+        done
+      workingDir: $(workspaces.source.path)
+
+    - name: oci-copy
+      image: quay.io/redhat-appstudio/buildah:v1.35.4@sha256:3d3575bb7d0df64abcf1f22f06e82101a945d03317db1f3caac12814f796d01c
+      computeResources:
+        limits:
+          memory: 1Gi
+        requests:
+          cpu: 250m
+          memory: 512Mi
+      securityContext:
+        capabilities:
+          add:
+            - SETFCAP
+      script: |
+        set -eu
+        set -o pipefail
+
+        buildah manifest create $IMAGE
+
+        for varfile in /var/workdir/vars/*; do
+          echo "Reading $varfile"
+          source $varfile
+
+          echo "Downloading $OCI_SOURCE to $OCI_FILENAME"
+          curl --silent --show-error --location $OCI_SOURCE -o $OCI_FILENAME
+
+          echo "Confirming that digest of $OCI_FILENAME matches expected $OCI_ARTIFACT_DIGEST"
+          echo "$OCI_ARTIFACT_DIGEST $OCI_FILENAME" | sha256sum --check
+
+          echo "Building manifest of type $OCI_ARTIFACT_TYPE from $OCI_FILENAME"
+          buildah manifest add $IMAGE --artifact --artifact-type $OCI_ARTIFACT_TYPE $OCI_FILENAME
+        done
+
+        echo "Pushing conents to $IMAGE"
+        buildah manifest push \
+            --digestfile $(workspaces.source.path)/image-digest \
+            --authfile $HOME/.docker/config.json \
+            --all \
+            $IMAGE
+
+        cat "$(workspaces.source.path)"/image-digest | tee $(results.IMAGE_DIGEST.path)
+        echo -n "$IMAGE" | tee $(results.IMAGE_URL.path)
+      volumeMounts:
+        - mountPath: /var/lib/containers
+          name: varlibcontainers
+      workingDir: $(workspaces.source.path)
+    - image: quay.io/konflux-ci/yq:latest@sha256:974dea6375ee9df561ffd3baf994db2b61777a71f3bcf0050c5dca91ac9b3430
+      name: sbom-generate
+      script: |
+        cat >sbom-cyclonedx.json <<EOL
+        {
+            "\$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
+            "bomFormat": "CycloneDX",
+            "specVersion": "1.5",
+            "version": 1,
+            "components": []
+        }
+        EOL
+
+        for varfile in /var/workdir/vars/*; do
+          echo "Reading $varfile"
+          source $varfile
+
+          ENCODED_URL=$(echo "${OCI_SOURCE}" | python3 -c 'import sys; import urllib.parse; print(urllib.parse.quote(sys.stdin.read().strip(), safe=":/"))')
+          purl="pkg:generic/${OCI_FILENAME}?download_url=${ENCODED_URL}&checksum=sha256:${OCI_ARTIFACT_DIGEST}"
```

Comment:
Just for correctness, we should encode the OCI_FILENAME as well

Otherwise LGTM

Reply:
Done in [bc71c7a](https://github.com/konflux-ci/build-definitions/pull/1077/commits/bc71c7a82da459851b3fc104a6f6075f9f1f5a8f).